Rumor.py

```diff
@@ -0,0 +1,57 @@
+from collections import deque
+from typing import List, Tuple
+
+def dfs(graph: List[List[int]], v: Tuple, visited: List[bool]) -> int:
+    idx: int = v[0]
+    bribe: int = v[1]
+
+    visited[idx] = True
+
+    for i in graph[idx]:
+        if not visited[i]:
+            #dfs(graph, (i, c[i-1]), visited)
+            return min(bribe, dfs(graph, (i, c[i]), visited) )
+
+    return bribe
+
+def bfs(graph, start: Tuple[int, int], visited):
+    queue = deque([start])
+
+    min_bribe = 1000000001
+    while queue:
+        v = queue.popleft()
+
+        idx = v[0]
+        bribe = v[1]
+
+        visited[idx] = True
+        min_bribe = min(min_bribe, bribe)
+
+        for character in graph[idx]:
+            if not visited[character]:
+                queue.append( (character, c[character]) )
+
+    return min_bribe
+
+# n : the number of chracters in Overcity
+# m : the number of pairs of friends
+n, m = map(int, input().split())
+c = list(map(int, input().split() ) )
+
+#graph: List[List[int]] = [ [] ] * (n+1)
+graph: List[List[int]] = [ [] for i in range(n+1) ]
```

Comment:
The line 41 was not worked.
It seems that the `n+1` allocated lists are shared the same memory address so that when I tried to append a value in a specific index of the lists, all lists in the `graph` is appended.